File: src/main/java/com/michalplachta/influencerstats/java/state/CollectionView.java
package com.michalplachta.influencerstats.java.state;

import com.michalplachta.influencerstats.java.core.Collection;

import java.util.Optional;
import java.util.concurrent.CompletableFuture;

public interface CollectionView {
    CompletableFuture<Optional<Collection>> fetchCollection(String id);
}

File: src/main/scala/com/michalplachta/influencerstats/state/CollectionView.scala
package com.michalplachta.influencerstats.state

import com.michalplachta.influencerstats.core.model.Collection
import simulacrum.typeclass

@typeclass
trait CollectionView[F[_]] {
  def fetchCollection(collectionId: String): F[Option[Collection]]
}
